Northstar.Client/mod/scripts/vscripts/ui/menu_mode_select.nut

```diff
@@ -1,86 +1,550 @@
+			case "hidden":
+			case "sns":
+			case "fw":
+			case "gg":
+			case "tt":
+			case "inf":
+			case "kr":
+			case "fastball":
+			case "hs":
+			case "ctf_comp":
+			case "tffa":
+				entry.category = eModeMenuModeCategory.CUSTOM
+				break
+		}
+
+		file.modes.append(entry)
+	}
+}
+
+int function SortModesAlphabetize( string a, string b )
+{
+	a = Localize( GetGameModeDisplayName( a ) )
+	b = Localize( GetGameModeDisplayName( b ) )
+
+	if ( a > b )
+		return 1
+
+	if ( a < b )
+		return -1
+
+	return 0
+}
+
+void function BuildSortedModesArray()
+{
+	file.sortedModes.clear()
+
+	// Build sorted list of categories
+	array<string> categories
+	for( int i = 0; i < eModeMenuModeCategory.SIZE; i++ )
+	{
+		if( file.searchEnum != -1 && file.searchEnum != i )
+			continue
+
+		categories.append( GetCategoryStringFromEnum( i ) )
+	}
+
+	categories.sort( SortStringAlphabetize )
+
+	// Build final list of mixed modes and categories
+	foreach( string category in categories )
+	{
+		// Build sorted list of modes in category
+		array<string> modes
+		foreach( ListEntry_t entry in file.modes )
+		{
+			int iCategory = entry.category
+			if( entry.mode in file.categoryOverrides )
+				iCategory = file.categoryOverrides[entry.mode]
+			
+			if( GetCategoryStringFromEnum( iCategory ) != category )
+				continue
+
+			string mode = entry.mode
+
+			if( file.searchString != "" && Localize(GetGameModeDisplayName(mode)).tolower().find(file.searchString.tolower()) == null )
+				continue
+
+			if( !modes.contains(mode) )
+				modes.append( mode )
+		}
+
+		modes.sort( SortModesAlphabetize )
+
+		if( modes.len() == 0 )
+			continue
+
+		// Add to final list we then display
+		file.sortedModes.append( category )
+		foreach( string mode in modes )
+			file.sortedModes.append( mode )
+	}
 }
 
+////////////////////////////
+// Slider
+////////////////////////////
+void function UpdateMouseDeltaBuffer( int x, int y )
+{
+	mouseDeltaBuffer.deltaX += x
+	mouseDeltaBuffer.deltaY += y
+
+	SliderBarUpdate()
+}
+
+void function FlushMouseDeltaBuffer()
+{
+	mouseDeltaBuffer.deltaX = 0
+	mouseDeltaBuffer.deltaY = 0
+}
+
+
+void function SliderBarUpdate()
+{
+	if( file.sortedModes.len() < MODES_PER_PAGE )
+		return
+
+	var sliderButton = Hud_GetChild( file.menu , "BtnModeListSlider" )
+	var sliderPanel = Hud_GetChild( file.menu , "BtnModeListSliderPanel" )
+	var movementCapture = Hud_GetChild( file.menu , "MouseMovementCapture" )
+
+	Hud_SetFocused( sliderButton )
+
+	float minYPos = -40.0 * ( GetScreenSize()[1] / 1080.0 )
+	float maxHeight = 596.0  * ( GetScreenSize()[1] / 1080.0 )
+	float maxYPos = minYPos - ( maxHeight - Hud_GetHeight( sliderPanel ) )
+	float useableSpace = ( maxHeight - Hud_GetHeight( sliderPanel ) )
+
+	float jump = minYPos - ( useableSpace / ( float( file.sortedModes.len() ) ) )
+
+	// got local from official respaw scripts, without untyped throws an error
+	local pos =	Hud_GetPos( sliderButton )[1]
+	local newPos = pos - mouseDeltaBuffer.deltaY
```

Comment:
could use `Hud_GetY` instead and type this but this code is 1 year old at this point so maybe not touch it